---
title: "Import data for Build Phyloseq"
author: "Hendra"
---


# install.packages("BiocManager")
library(BiocManager)
# install("phyloseq")
# install("ggtree")
library(phyloseq)
library(Biostrings)
# ?read.delim
# ?read.table
# ?read.csv
library(ape)
# ?read.tree
library(ggtree)
# ?read.tree
# ?read_tre
# ?readDNAStringSet
library(plyr)
library(reshape2)  
library(data.table) 

#---Input Data----#

otu = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/OTU table new.txt",row.names = 1)
tax = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/Taxonomy table.txt",row.names = 1)
map = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/rice-seed-metadataR.txt",row.names = 1)
map$ID = row.names(map)

head(map)
head(tax)

----------------------------------------------------------------------------------------------------
library(ggtree)
tree  = read_tree("D:/S3-KU/Rice seed analysis/Phyloseq/Phylogenetic tree new.nwk")
library(Biostrings)
rep = readDNAStringSet("D:/S3-KU/Rice seed analysis/Phyloseq/sequences data.fasta")
------------------------------------------------------------------------------------------------------
library(phyloseq)
psseed = phyloseq(sample_data(map),
              otu_table(as.matrix(otu), taxa_are_rows=TRUE), 
              tax_table(as.matrix(tax)),
              phy_tree(tree),
              refseq(rep)
              )
psseed

saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-all.rds")
--------------------------------------------------------------------------------------------------------
#------Sub set Metadata-------#

psorigin <- subset_samples(psseed,  Source == "Origin")

saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-origin.rds")

pscultur <- subset_samples(psseed,  Source == "Culturable")

saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-cultur.rds")
-----------------------------------------------------------------------------------------------------------

#--------Change to ASV conversion----------#

asv_names <- paste0("ASV_", seq(1:ntaxa(psseed)))  # Make the new name
# Save the original OTU name
old_names <- taxa_names(psseed)

# Rename components in each phyloseq object
taxa_names(psseed) <- asv_names  # Change name

# Rename all components in the phyloseq object
otu_table(psseed) <- otu_table(psseed)[asv_names, ]  # Ensure correct order
tax_table(psseed) <- tax_table(psseed)[asv_names, ]
phy_tree(psseed) <- prune_taxa(asv_names, phy_tree(psseed))  # if have tree
refseq(psseed) <- refseq(psseed)[asv_names]

new_refseq <- refseq(psseed)  # pick old refseq 
names(new_refseq) <- asv_names  # Change name to ASV
new_refseq <- Biostrings::DNAStringSet(new_refseq)  # reset allign to format

Biostrings::writeXStringSet(new_refseq, filepath = "new_refseq_output.fasta")

psseed_asv <- phyloseq(
  sample_data(psseed),
  otu_table(psseed),
  tax_table(psseed),
  phy_tree(psseed),
  new_refseq  # Input the new refseq
)

psseed_asv
----------------------------------------------------------------------------------------------------
# all(taxa_names(psseed_asv) == names(refseq(ps_asv)))  # TRUE
# tax_table(ps_asv)
# otu_table(psseed_asv)
# saveRDS(ps_asv,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-all.rds")
# psseed_asv_origin <- subset_samples(ps_asv,  Source == "Origin")
# saveRDS(psseed_asv_origin,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-origin.rds")
# psseed_asv_cultur <- subset_samples(psseed_asv,  Source == "Culturable")
# saveRDS(psseed_asv_cultur,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-cultur.rds")
```

