---
title: "Import data for Build Phyloseq-Tao Wen 2023"
author: "Hendra"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
# install.packages("BiocManager")
library(BiocManager)
# install("phyloseq")
# install("ggtree")
library(phyloseq)
library(Biostrings)
# ?read.delim
# ?read.table
# ?read.csv
library(ape)
# ?read.tree
library(ggtree)
# ?read.tree
# ?read_tre
# ?readDNAStringSet
library(plyr)

library(reshape2)  # Untuk melt dari reshape2
# atau
library(data.table)  # Untuk melt dari data.table
```

#---Input Data----#

```{r}
otu = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/OTU table new.txt",row.names = 1)
tax = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/Taxonomy table.txt",row.names = 1)
map = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/rice-seed-metadataR.txt",row.names = 1)
map$ID = row.names(map)

head(map)
```

```{r}
head(tax)
```


```{r}
library(ggtree)
tree  = read_tree("D:/S3-KU/Rice seed analysis/Phyloseq/Phylogenetic tree new.nwk")

library(Biostrings)
rep = readDNAStringSet("D:/S3-KU/Rice seed analysis/Phyloseq/sequences data.fasta")
```

```{r}
library(phyloseq)

psseed = phyloseq(sample_data(map),
              otu_table(as.matrix(otu), taxa_are_rows=TRUE), 
              tax_table(as.matrix(tax)),
              phy_tree(tree),
              refseq(rep)
              )
psseed
```

```{r}
saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-all.rds")
```

#------SUB-SET METADATA-------#

```{r}
psorigin <- subset_samples(psseed,  Source == "Origin")
```

```{r}
saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-origin.rds")
```


```{r}
pscultur <- subset_samples(psseed,  Source == "Culturable")
```

```{r}
saveRDS(ps,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeq-cultur.rds")
```


#--------Change to ASV conversion----------#

```{r}
asv_names <- paste0("ASV_", seq(1:ntaxa(psseed)))  # Make the new name
```

```{r}
# Save the original OTU name
old_names <- taxa_names(psseed)

# Rename components in each phyloseq object
taxa_names(psseed) <- asv_names  # Change name

# Rename all components in the phyloseq object
otu_table(psseed) <- otu_table(psseed)[asv_names, ]  # Ensure correct order
tax_table(psseed) <- tax_table(psseed)[asv_names, ]
phy_tree(psseed) <- prune_taxa(asv_names, phy_tree(psseed))  # if have tree
refseq(psseed) <- refseq(psseed)[asv_names]
```

```{r}
new_refseq <- refseq(psseed)  # pick old refseq 
names(new_refseq) <- asv_names  # Change name to ASV
new_refseq <- Biostrings::DNAStringSet(new_refseq)  # reset allign to format
```

```{r}
Biostrings::writeXStringSet(new_refseq, filepath = "new_refseq_output.fasta")
```

```{r}
psseed_asv <- phyloseq(
  sample_data(psseed),
  otu_table(psseed),
  tax_table(psseed),
  phy_tree(psseed),
  new_refseq  # Input the new refseq
)

psseed_asv
```

```{r}
all(taxa_names(psseed_asv) == names(refseq(ps_asv)))  # Harus TRUE
```

```{r}
tax_table(ps_asv)
```


```{r}
otu_table(psseed_asv)
```


```{r}
saveRDS(ps_asv,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-all.rds")
```

```{r}
psseed_asv_origin <- subset_samples(ps_asv,  Source == "Origin")
```

```{r}
saveRDS(psseed_asv_origin,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-origin.rds")
```

```{r}
psseed_asv_cultur <- subset_samples(psseed_asv,  Source == "Culturable")
```


```{r}
saveRDS(psseed_asv_cultur,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Phyasv-cultur.rds")
```






## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


#-----------------------------GANTI TAXA ASV-----------------#

```{r}
otu = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/ASVs_table.txt",row.names = 1)
tax = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/Taxonomy_ASVss.txt",row.names = 1)
map = read.delim("D:/S3-KU/Rice seed analysis/Phyloseq/rice-seed-metadataR.txt",row.names = 1)
map$ID = row.names(map)

head(map)
```


```{r}
library(phyloseq)

pso = phyloseq(sample_data(map),
              otu_table(as.matrix(otu), taxa_are_rows=TRUE), 
              tax_table(as.matrix(tax))
              )
pso
```

```{r}
psori <- subset_samples(pso,  Source == "Origin")
```

```{r}
saveRDS(psori,"D:/S3-KU/Rice seed analysis/Phyloseq/New Phylo/Physeqasvori.rds")
```





